test/unit/authentication/custom_credential_manager_test.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const assert = require('assert');
+const Util = require('../../../lib/util');
+const { randomUUID } = require('crypto');
+const GlobalConfig = require('../../../lib/global_config');
+const JsonCredentialManager = require('../../../lib/authentication/secure_storage/json_credential_manager');
+const host = 'mock_host';
+const user = 'mock_user';
+const credType = 'mock_cred';
+const key = Util.buildCredentialCacheKey(host, user, credType);
+const randomPassword = randomUUID();
+const defaultCredentialManager = new JsonCredentialManager();
+const mockCustomCrednetialManager = {
+  read: function () {
+    return 'mock_token';
+  },
+  write: function () {
```

Comment:
don't we expect these functions to return promise? (or being async?)

Reply:
There are two test case scenarios. 1. Sync 2. Async.
You can find the async testing case at the bottom: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/820/files#diff-c92bed55d478910c2b3d0fa5930295b793b9d866ee42e7411ad6ac4fc15531adR66